Library/Homebrew/utils/tty.rb

```diff
@@ -0,0 +1,45 @@
+module Tty
+  module_function
+
+  def strip_ansi(str)
+    str.gsub(/\033\[\d+(;\d+)*m/, "")
+  end
+
+  def width
+    `/usr/bin/tput cols`.strip.to_i
+  end
+
+  def truncate(str)
+    (w = width).zero? ? str.to_s : str.to_s[0, w - 4]
```

Comment:
`str` -> `string`, use a longer variable for `w`, use `if` over ternary, use a more self-documenting function for `[]`